### ISSUE TITLE
`error_types` option not respected on `captureLastError`

### ISSUE THREAD

**Issue by GrimmGringo** [getsentry/sentry-php#1115]
TL;DR: fix is attached

Here's the problem:
```
    Sentry\init([
        'dsn' => 'xxx',
        'error_types' => E_ERROR|E_WARNING, // should ignore notices such as array indexes not existing
```
At the very bottom of my PHP application, I call this:
`Sentry\captureLastError();`

Now, if my code has $myvar['fakeindex'] (where fakeindex is never set), it is still going to throw an alert for that notice. E_NOTICE = 8, but E_ERROR|E_WARNING = 3. I have specified in my error_types to ONLY show error or warnings. Sentry is not respecting that, and I'm getting alerts for array indexes that dont exist. 

Took me a lot of digging to figure this one out, learned way more about how Sentry for PHP works than I really wanted to. 

But anyway, the fix is for Client.php, attached in this zip.
[Client.zip](https://github.com/getsentry/sentry-php/files/5362248/Client.zip)

Basically if you look in the prepareEvent function, at line 217, I added this code:

```
                if ($hint->exception instanceof \ErrorException) {
                    if ( $hint->exception->getSeverity() > $this->options->getErrorTypes() ) {
                        return null;
                    }
                }
```

If the "exception"  severity is greater than the error_types you specified, it will return null.

To better explain, if E_NOTICE = 8 and E_ERROR|E_WARNING = 3:
`if ( 8 > 3 ) { return null }`
Because 8 is greater than 3, it will return null and not error on a notice. 

Now, if you have it set to E_ALL (which is 32767), it WOULD alert that index issue:
`if ( 8 > 32767 ) { return null }`

Because 8 is NOT greater than 32767, so it will not return null -- it will proceed on as intended.

**Comment by stayallive** [comment]
Hi @GrimmGringo, thank you for your report but let me explain to you what you missed.

The `error_types` option only has effect on our automatic error listener, not on errors manually reported, you can see the code that checks the option to only report the enabled errors here: https://github.com/getsentry/sentry-php/blob/f09173d6100a0fd802b666b8a00c0bd2a1b472a3/src/Integration/ErrorListenerIntegration.php#L38-L40.

But you are calling `captureLastError` which does not work with `error_types`. It will always capture the last error. This is intended and you should not call that code, there is no need too.

Now you mention also that you compare the `error_types` value as a number, it is not a number, it can be represented as a number but it's a bit mask. I will not try to explain that to you here since the internet can explain how a bit mask works a lot better but the code in the `ErrorListenerIntegration` (`$client->getOptions()->getErrorTypes() & $exception->getSeverity()`, notice the single `&`) is the correct way to test if the severity is included in the bit mask of the `error_types` option.

---

So long story short, remove the `captureLastError` and the errors should be reported as you would expect 👍

**Comment by GrimmGringo** [comment]
Hi @stayallive 

Thank you for taking the time to reply back in full, appreciate that!

I understand removing the Sentry\captureLastError() makes it work as I'm expecting, but that function itself is what I'm having an issue with. If you have explicitly set error_types to a value, why would you want Sentry\captureLastError() to give you any error at all?

Different mindset I suppose on this, but as an actual end-user, if I want to capture the last error, I only want to know about the last error that fit my error_types filtering. 

I still think this is a bug, or at a minimum should be optional to enforce it behaving as I would expect it to. Surely there are others in my situation who would like this behavior?

**Comment by stayallive** [comment]
@GrimmGringo I see, may I ask what your use case is for calling `captureLastError` at the end of your script then? Because I don't see how it's useful considering that error was already captured by the `ErrorListenerIntegration` if it matched your `error_types` config.

So I'm trying to see why you would want to call `captureLastError` and have it respect that option because I think the idea is that you capture every single error matching your `error_types` and if you wanted to capture an error that is not in your `error_types` you can use `captureLastError` to do that.

**Comment by GrimmGringo** [comment]
Hi, 

I'm no longer calling captureLastError manually, since it's unnecessary. However to me, if you are specifying error_types in the init, then that should mean that you only want it to capture those. 

So if I have 'error_types' => E_ERROR | E_DEPRECATED  , then I'm not wanting to see warnings. Right now, that isn't how this is working. If you put error_reporting(E_ALL) and you keep error_types as E_ERROR, but there's a warning, then Sentry PHP is going to log it. To me that's incorrect.

**Comment by ste93cry** [comment]
I'm closing this issue because it's quite old and there are no plans for now to revise the decision of changing the current behaviour of `captureError()`. Nothing is written in stone though, and if in the future many people will ask this we will of course re-evaluate the situation
